akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -73,8 +76,21 @@ trait AskSupport {
    *
    */
   def ask(actorRef: ActorRef, message: Any)(implicit timeout: Timeout): Future[Any] =
-    actorRef.internalAsk(message, timeout, ActorRef.noSender)
+    actorRef.internalAsk(_ ⇒ message, timeout, ActorRef.noSender)
   def ask(actorRef: ActorRef, message: Any, sender: ActorRef)(implicit timeout: Timeout): Future[Any] =
+    actorRef.internalAsk(_ ⇒ message, timeout, sender)
+
+  /**
+   * A variant of `ask()` which allows to explicitly provide a reference to the actor
+   * which the result needs to be sent to in a message:
+   *
+   * {{{
+   *   ask(worker, respondTo => Message(respondTo))(timeout)
+   * }}}
+   */
+  def ask(actorRef: ActorRef, message: ActorRef ⇒ Any)(implicit timeout: Timeout): Future[Any] =
```

Comment:
This introduces a tricky question about Java API:
- We should provide Java API for it, since on Scala 2.11 it's not nice to use this ask, it would have to be `ask(ActorRef, japi.Function1)`
- Scala 2.12 comes out, and `ActorRef => Any` can now be used from Java 8 as `e -> m`, won't java8 freak out that it's ambiguous which method we wanted - the "java api" (added for Scala 2.11 + JDK7) or the "scala api, which now works with Java 8"?

So... Since 2.4 requires JDK8, Java users will most likely be able to bump from 2.11 to 2.12 since they don't depend on Scala compat (I'd hope), and use the new API via the scala Functions. Summing up - it's best we do not add specific Java API for this method? Unless I missed something.